matscipy/neighbours.py

```diff
@@ -150,6 +150,22 @@ def neighbour_list(quantities, a, cutoff):
                                   np.arange(len(a) + 1)),
                                  shape=(3 * len(a), 3 * len(a)))
 
+
+  i_n, j_n, dr_nc, abs_dr_n = neighbour_list('ijDd', atoms, dict)
+e_nc = (dr_nc.T/abs_dr_n).T
+    D_ncc = -(dde_n * (e_nc.reshape(-1,3,1) * e_nc.reshape(-1,1,3)).T).T
+    D_ncc += -(de_n/abs_dr_n * (np.eye(3, dtype=e_nc.dtype) - (e_nc.reshape(-1,3,1) * e_nc.reshape(-1,1,3))).T).T
+
+    D = bsr_matrix((D_ncc, j_n, first_i), shape=(3*nat,3*nat))
+
+    Ddiag_icc = np.empty((nat,3,3))
+    for x in range(3):
+        for y in range(3):
+            Ddiag_icc[:,x,y] = -np.bincount(i_n, weights = D_ncc[:,x,y])
+
+    D += bsr_matrix((Ddiag_icc,np.arange(nat),np.arange(nat+1)), shape=(3*nat,3*nat))
```

Comment:
Please try to follow pep-8 throughout. Here: Spaces after ',' and around binary operators, e.g. '+'.

What happens if bsr_matrix (from scipy?) is not available?

Reply:
Sorry, I see this is an example. Please still use pep-8